addons/product/static/src/product_catalog/order_line/order_line.xml

```diff
@@ -30,7 +30,9 @@
                 <input class="o_input form-control border text-center text-bg-light"
                         type="number"
                         t-att-value="quantity"
-                        t-on-change="this.env.setQuantity"/>
+                        t-on-change="this.env.setQuantity"
+                        t-on-click="(event) => event.target.select()"
```

Comment:
I'd rather use `focus` than `click` events to support scenarios where users do not click on the element (like using `tab`).
```suggestion
                        t-on-focus="(ev) => ev.target.select()"
```
---
nitpicking: it would be event better IMO to do this in a small method in the js file
```suggestion
                        t-on-focus="_onFocus"
```
```js
_onFocus(ev) {
        ev.target.select();
    }
```